source/DSCResources/MSFT_xDnsRecordMx/MSFT_xDnsRecordMx.psm1

```diff
@@ -0,0 +1,286 @@
+    .PARAMETER Zone
+        Specifies the name of a DNS zone.
+
+    .PARAMETER Target
+        Specifies the Target Hostname or IP Address.
+
+    .PARAMETER Priority
+        Specifies the Priority value of the MX record.
+
+    .PARAMETER TTL
+        Specifies the TTL value of the MX record. Value must be in valid TimeSpan format.
+
+    .PARAMETER DnsServer
+        Name of the DnsServer to create the record on.
+
+    .PARAMETER Ensure
+        Whether the host record should be present or removed.
+#>
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Name,
+
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Zone,
+
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Target,
+
+        [Parameter()]
+        [System.UInt16]
+        $Priority,
+
+        [Parameter()]
+        [ValidateScript({$ts = New-TimeSpan; [system.timespan]::TryParse($_, [ref]$ts)})]
+        [System.String]
+        $TTL,
+
+        [Parameter()]
+        [System.String]
+        $DnsServer = "localhost",
+
+        [Parameter()]
+        [ValidateSet('Present','Absent')]
+        [System.String]
+        $Ensure = 'Present'
+    )
+
+    $DNSParameters = @{
+        Name         = $Name
+        ZoneName     = $Zone
+        ComputerName = $DnsServer
+    }
+
+    if ($Ensure -eq 'Present')
+    {
+        $DNSParameters.Add('Mx',$true)
+        $DNSParameters.Add('MailExchange',$Target)
+        $DNSParameters.Add('Preference',$Priority)
+
+        if ($null -ne $TTL)
+        {
+            $DNSParameters.Add('TimeToLive', $TTL)
+        }
+
+        Write-Verbose -Message ($script:localizedData.CreatingDnsRecordMessage -f 'MX', $Target, $Zone, $DnsServer)
+        Add-DnsServerResourceRecord @DNSParameters
+    }
+    elseif ($Ensure -eq 'Absent')
+    {
+        $DNSParameters.Add('Force',$true)
+        $DNSParameters.Add('RRType','Mx')
+
+        Write-Verbose -Message ($script:localizedData.RemovingDnsRecordMessage -f 'MX', $Target, $Zone, $DnsServer)
+        Remove-DnsServerResourceRecord @DNSParameters
+    }
+} #end function Set-TargetResource
+
+<#
+    .SYNOPSIS
+        This will return whether the resource is in desired state.
+
+    .PARAMETER Name
+        Specifies the name of the DNS server resource record object. For records in the apex of the domain, use a period.
+
+    .PARAMETER Zone
+        Specifies the name of a DNS zone.
+
+    .PARAMETER Target
+        Specifies the Target Hostname or IP Address.
+
+    .PARAMETER Priority
+        Specifies the Priority value of the MX record.
+
+    .PARAMETER TTL
+        Specifies the TTL value of the MX record. Value must be in valid TimeSpan format.
+
+    .PARAMETER DnsServer
+        Name of the DnsServer to create the record on.
+
+    .PARAMETER Ensure
+        Whether the host record should be present or removed.
+#>
+function Test-TargetResource
```

Comment:
Currently in Test you are only seeing if $target is correct. Since you can set Priority and TTL should you be checking those settings if they are set? Also, if someone changes a value and Set would run again you are only doing Add-DnsServerResourceRecord which will fail if the record already exists.

Reply:
This should be fixed now. Not quite sure how to test if Set-DnsServerResourceRecord is being called because I'm not able to mock the input parameter type CimObject.